test/test_cuda.py

```diff
@@ -1717,24 +1752,33 @@ def test_stream_event_nogil(self):
                           TestCuda._event_wait]:
             p2c = queue.Queue()
             c2p = queue.Queue()
+            e_tik = torch.cuda.Event(enable_timing=True)
+            e_tok = torch.cuda.Event(enable_timing=True)
 
             t = threading.Thread(
                 target=TestCuda._test_stream_event_nogil,
-                args=(sync_func, p2c, c2p))
+                args=(self, sync_func, p2c, c2p))
             t.daemon = True
             t.start()
 
             c2p.get()
-            tik = time.time()
             with torch.cuda.device('cuda:0'):
+                e_tik.record()
                 p2c.put(0)
-                sync_func(50000000)
-
-            t.join()
-            tok = time.time()
-
-            self.assertGreater(tok - tik, 0.05)
-            self.assertLess(tok - tik, 0.1)
+                parent_time = sync_func(self, TestCuda.FIFTY_MIL_CYCLES)
+                child_time = c2p.get()
+                e_tok.record()
+                e_tok.synchronize()
+                total_time = e_tik.elapsed_time(e_tok)
+
+            # Without GIL, the two 50ms synchronization can overlap, and hence
```

Comment:
I added an assumption at the end: "assuming 1 cycle is 1 nano sec". Will this do it?

Reply:
Oh I see. But I don't think that it is a reasonable assumption. I think the key here is that we can't rely on any computed expected time duration because that 1. the max clock rate is card-dependent, and 2. the actual clock rate depends on many factors like power.

Reply:
I see. Will make the change, thanks!